<a href="https://colab.research.google.com/github/alexDx12/gb_statistics/blob/lesson_6/lesson_6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Теория вероятностей и математическая статистика

## Урок 6

## Взаимосвязь величин. Показатели корреляции. Корреляционный анализ. Проверка на нормальность

### Домашнее задание

### Задача 1

__Постановка задачи__

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy.
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

__Решение__

Импорт библиотек и скриптов:

In [ ]:
import numpy as np

Значения величины заработной платы заемщиков (zp) и их поведенческого кредитного скоринга (ks) исходя из постановки задачи:

In [ ]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

Выборочное среднее величины заработной платы заемщиков (M_zp):

In [ ]:
M_zp = zp.mean()
M_zp

101.4

Выборочное среднее значения поведенческого кредитного скоринга заемщиков (M_ks):

In [ ]:
M_ks = ks.mean()
M_ks

709.9

Ковариация, расчитываемая с помощью элементарных действий (cov):

In [ ]:
cov = ((zp - M_zp) * (ks - M_ks)).sum() / (zp.shape[0] - 1)
round(cov, 2)

10175.38

Ковариация, расчитываемая с помощью библиотеки numpy (np_cov):

In [ ]:
np_cov = np.cov(zp, ks, ddof=1)[1][0]
round(np_cov, 2)

10175.38

Проверка равенства значений cov и np_cov (True - значения равны, False - значения не равны):

In [ ]:
round(cov, 2) == round(np_cov, 2)

True

Коэффициент корреляции Пирсона, рассчитанный с помощью ковариации и среднеквадратичных отклонений признаков (corr):

In [ ]:
corr = cov / (zp.std(ddof=1) * ks.std(ddof=1))
round(corr, 2)

0.89

Коэффициент корреляции Пирсона, рассчитанный с помощью библиотеки numpy (np_corr):

In [ ]:
np_corr = np.corrcoef(zp, ks)[1][0]
round(np_corr, 2)

0.89

Проверка равенства значений corr и np_corr (True - значения равны, False - значения не равны):

In [ ]:
round(corr, 2) == round(np_corr, 2)

True

Ответ: ковариация значений равна 10175.38, коэффициент корреляции Пирсона равен 0.89.

### Задача 2

__Постановка задачи__

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

__Решение__

Импорт библиотек и скриптов:

In [ ]:
import numpy as np
from scipy import stats

Значения выборки IQ студентов (iq_sample) и надёжности (p) исходя из постановки задачи:

In [ ]:
iq_sample = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
p = 0.95

Выборочное среднее выборки IQ студентов (mean):

In [ ]:
mean = iq_sample.mean()
mean

118.1

Несмещённая оценка стандатного отклонения выборки IQ студентов (std):

In [ ]:
std = iq_sample.std(ddof=1)
std

10.54566788359614

Уровень значимости (alpha):

In [ ]:
alpha = 1 - p
alpha

0.050000000000000044

Объём выборки (n):

In [ ]:
n = len(iq_sample)
n

10

Квантили распределения (t1, t2):

In [ ]:
t1 = stats.t.ppf(alpha / 2, df=n - 1)
t2 = stats.t.ppf(1 - alpha / 2, df=n - 1)
t1, t2

(-2.2621571627409915, 2.2621571627409915)

Доверительный интервал (conf):

In [ ]:
conf = (mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))
conf

(110.55608365158724, 125.64391634841274)

Ответ: доверительный интервал составляет от 110.556 до 125.644.

### Задача 3

__Постановка задачи__

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

__Решение__

Импорт библиотек и скриптов:

In [ ]:
from scipy import stats

Значения дисперсии генеральной совокупности (variance), объёма выборки (n), среднего выборочного (mean) и надёжности (p) исходя из постановки задачи:

In [ ]:
variance = 25
n = 27
mean = 174.2
p = 0.95

Уровень значимости (alpha):

In [ ]:
alpha = 1 - p
alpha

0.050000000000000044

Квантили распределения (t1, t2):

In [ ]:
t1 = stats.norm.ppf(alpha / 2)
t2 = stats.norm.ppf(1 - alpha / 2)
t1, t2

(-1.959963984540054, 1.959963984540054)

Доверительный интервал (conf):

In [ ]:
conf = (mean + t1 * std / np.sqrt(n), mean + t2 * std / np.sqrt(n))
conf

(170.22222422220577, 178.1777757777942)

Ответ: доверительный интервал составляет от 170.222 до 178.178.

### Задача 4

__Постановка задачи__

Выберите тему для проектной работы по курсу Теории вероятностей и математической статистики
и напишите ее в комментарии к Практическому заданию.

__Решение__

Тема проектной работы: Exploratory Data Analysis (EDA) на примере задачи "Titanic - Machine Learning from Disaster" платформы kaggle (https://www.kaggle.com)".